### DISCUSSION TITLE
Pytest test, headless mode differences?

### DISCUSSION THREAD

**Discussion by iceman8420** [seleniumbase/SeleniumBase#2569] [Q&A] [answered]
Hi all,
Thanks again for the great module and community. 
Learning a lot from here and making improvements with my test cases all of the time!

Running into a weird issue.
My pytest works great with an interactive window, but not when using headless or headless2 options. 
The content from the test site is not loading appropriately.

My tests work just fine here:

`#BaseCase.main(__name__, __file__,'--uc')`

But not launching like this (or --headless):
`BaseCase.main(__name__, __file__, "--uc", "--headless2")`

Any behavioral differences or suggestions on what to look into?

**Comment by mdmintz** [comment, marked as answer]
The `BaseCase.main(__name__, __file__)` line is only read when running the script with pure `python`.
(It's there in case people forget to use `pytest` instead.)

If you're running the script directly with `pytest`, add the options you need as command-line options. Eg:

```bash
pytest --uc --headless2
```

**Reply by iceman8420** [reply]
Thanks! Don't I feel dumb....
